lib/collection/src/shards/local_shard.rs

```diff
@@ -178,22 +178,21 @@ impl LocalShard {
 
         for entry in segment_dirs {
             let segments_path = entry.unwrap().path();
-            if segments_path.ends_with("deleted") {
-                remove_dir_all(&segments_path).await.map_err(|_| {
-                    CollectionError::service_error(format!(
-                        "Can't remove marked-for-remove segment {}",
-                        segments_path.to_str().unwrap()
-                    ))
-                })?;
-                continue;
-            }
```

Comment:
@generall Can you, please, explain how this one was causing panic? Seems like this should have worked more or less the same as the new code. 🤔

Reply:
`ends_with` doesn't do what you would expect it to do. See docs

Reply:
Got it. 👌

```
Determines whether child is a suffix of self.

Only considers whole path components to match.
```